web/packages/teleport/src/components/HeadlessRequestDialog/HeadlessRequestDialog.tsx

```diff
@@ -25,39 +25,75 @@ import Dialog, {
 import { Danger } from 'design/Alert';
 import { ButtonPrimary, ButtonSecondary, Text } from 'design';
 
-export default function HeadlessSsoDialog({
+export default function HeadlessRequestDialog({
   ipAddress,
   onAccept,
   onReject,
   errorText,
 }: Props) {
+  const dialogContent = () => {
```

Comment:
I'm not a fan of splitting a component into parts like this, mostly because you can no longer read the component top to bottom to understand the structure of what is going to end up being rendered.

The conditionals could be inlined which IMHO would do a better job of showing what goes in the dialog content and what goes in the footer:

```typescript
export default function HeadlessRequestDialog({
  ipAddress,
  onAccept,
  onReject,
  errorText,
}: Props) {
  return (
    <Dialog dialogCss={() => ({ width: '400px' })} open={true}>
      <DialogHeader style={{ flexDirection: 'column' }}>
        <DialogTitle textAlign="center">
          Host {ipAddress} wants to execute a command
        </DialogTitle>
      </DialogHeader>
      <DialogContent mb={6}>
        {errorText && (
          <Danger mt={2} width="100%">
            {errorText}
          </Danger>
        )}
        <Text textAlign="center">
          {errorText ? (
            <>
              The requested session doesn't exist or is invalid. Please generate
              a new request.
              <br />
              <br />
              You can close this window.
            </>
          ) : (
            <>
              Someone has initiated a command from {ipAddress}. If it was not
              you, click reject and contact your administrator.
              <br />
              <br />
              If it was you, please use your hardware key to approve.
            </>
          )}
        </Text>
      </DialogContent>
      <DialogFooter textAlign="center">
        {!errorText && (
          <>
            <ButtonPrimary onClick={onAccept} autoFocus mr={3} width="130px">
              Approve
            </ButtonPrimary>
            <ButtonSecondary onClick={onReject}>Reject</ButtonSecondary>
          </>
        )}
      </DialogFooter>
    </Dialog>
  );
}
```

Reply:
Not React Dev. Applying what people tells me is better 😅